doc/application/application.rst

```diff
@@ -369,6 +369,75 @@ again.
 
 .. _application_debugging:
 
+
+Custom Board Definition
+***********************
+
+In cases where the board or platform you are developing for is not yet supported
+by Zephyr you can add the board definition to your application and build for
+this board without having to add it to the Zephyr tree.
+
+The structure needed to support an out-of-tree board is similar to how boards
+are maintained in Zephyr making it easy to upstream a board to Zephyr once the
+initial development is done out of Zephyr.
+
+Add the custom board to your application using the following structure:
+
+.. code-block:: console
+
+   boards/
+   CMakeLists.txt
+   prj.conf
+   README.rst
+   src/
+
+where the boards directory hosts the board you are building for:
+
+.. code-block:: console
+
+   .
+   ├── boards
+   │   └── x86
+   │       └── my_custom_board
+   │           ├── doc
+   │           │   └── img
+   │           └── support
+   └── src
+
+
+Documentation (under doc/) and support files (under support/) are optional, but
```

Comment:
Format the folder names using:

    Documentation (under ``doc/``) and support files (under ``support/``)